relations/src/hasher.rs

```diff
@@ -0,0 +1,83 @@
+use halo2_base::{
+    gates::GateChip, poseidon::hasher::PoseidonHasher, utils::BigPrimeField, AssignedValue, Context,
+};
+
+use crate::{account::CircuitAccount, note::CircuitNote};
+
+pub trait InnerHasher<F: BigPrimeField> {
+    fn hash_account(
+        &mut self,
+        ctx: &mut Context<F>,
+        gate: &GateChip<F>,
+        account: &impl CircuitAccount<F>,
+    ) -> AssignedValue<F>;
+
+    fn hash_note(
+        &mut self,
+        ctx: &mut Context<F>,
+        gate: &GateChip<F>,
+        note: &CircuitNote<F>,
+    ) -> AssignedValue<F>;
+}
+
+impl<F: BigPrimeField, const T: usize, const RATE: usize> InnerHasher<F>
+    for PoseidonHasher<F, T, RATE>
+{
+    fn hash_account(
+        &mut self,
+        ctx: &mut Context<F>,
+        gate: &GateChip<F>,
+        account: &impl CircuitAccount<F>,
+    ) -> AssignedValue<F> {
+        let account_array = account.to_array();
+        self.hash_fix_len_array(ctx, gate, &account_array)
+    }
+
+    fn hash_note(
+        &mut self,
+        ctx: &mut Context<F>,
+        gate: &GateChip<F>,
+        note: &CircuitNote<F>,
+    ) -> AssignedValue<F> {
+        let note_array = note.to_array();
+        self.hash_fix_len_array(ctx, gate, &note_array)
+    }
+}
+
+// #[cfg(test)]
+pub mod tests {
```

Comment:
what is the purpose of this module?

Reply:
It was `utils for testing only`, but now file no longer exist, so done